02_bubble/willshen8/bubble_test.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := "[1 2 3 5]"
+	actual := buf.String()
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main(), expected = %v, actual = %v", expected, actual)
+	}
+}
+
+func TestBubbleAndInsertionSort(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    []int
+		expected []int
+	}{
+		{name: "Empty slice", input: []int{}, expected: []int{}},
```

Comment:
Great test coverage and incorporating both sorts in the same test. We will do suite testing when we get into the rest api and you will see how you can run the same test on multiple implementations of a particular function/interface.